packages/editor/src/components/provider/use-block-editor-settings.native.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import useBlockEditorSettings from './use-block-editor-settings.js';
+
+function useNativeBlockEditorSettings( settings, hasTemplate ) {
```

Comment:
The purpose of this hook is to add custom code to the `useBlockEditorSettings` hook, which will be only executed in the native version of the editor.

Reply:
Thanks for the context here. I can see that `useNativeBlockEditorSettings` is utilizing the web variant to create the `editorSettings` and then based on if support for reusable block is enabled or not based on the site type, we are then querying the `coreStore` for the said blocks.

Reply:
Yeah exactly, the main difference between this version and the web is that `capabilities` is only available in native. This is used, as you commented, to control whether it should query the reusable blocks.